pos_sale_order/models/sale_order.py

```diff
@@ -112,8 +112,17 @@ def _compute_pos_payment(self):
                 record.pos_amount_to_pay = 0
                 record.pos_payment_state = "none"
             else:
-                residual = record.amount_total - sum(
-                    record.mapped("payment_ids.amount")
+                residual = (
+                    record.amount_total
+                    # Substract the amount already paid on order
+                    - sum(record.mapped("payment_ids.amount"))
+                    # Substract the amount already paid on invoice
+                    - sum(
+                        record.invoice_ids.filtered(
```

Comment:
I think the @api.depends should depend on "invoice_ids.payment_state" as the field is stored